### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by mesudpasic**

### CODE REVIEW

File: Module.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+version: 1.0.0
+name: OPCUASubscribe
```
Comment (kubagrzelak) [inline Module.yaml]: add spacebar in the name `OPC UA Subscribe`

File: Module.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+description: OPC UA Subscribe connects to OPC UA Server and listens for incoming data messages and forwards the payload to next module.
+icon: https://icons-020-demo.s3.eu-central-1.amazonaws.com/mi_OPCUA_input.png
```
Comment (kubagrzelak) [inline Module.yaml]: so it is type `process` module but you use icon for `input` (so it has a different colour). Can you ask Christina for `process` type icon for this module as it looks better in UI?

File: Module.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+type: process
+tags:
+  - OPC UA Subscribe
+  - process
+categories:
+  - id: category
+    name: Process
```
Comment (kubagrzelak) [inline Module.yaml]: We don't have a category `Process`. I would suggest `Policy` category.

File: Module.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
+version: 1.0.0
+name: OPCUASubscribe
+description: OPC UA Subscribe connects to OPC UA Server and listens for incoming data messages and forwards the payload to next module.
```
Comment (kubagrzelak) [inline Module.yaml]: If this module subscribes to a server, why would it be of type `process`? If you build a pipeline with this module, do you MUST use another module before this one? If not, then it should be `input` module

File: Module.yaml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    value: ''
+    default: 10
+    type: integer
+    name: Requested max keep alive count
+  - key:  MAX_NOTIFICATIONS_PER_PUBLISH
+    description: maxNotificationsPerPublish parameter
+    value: ''
+    default: 100
+    type: integer
+    name: max notifications per publish
```
Comment (kubagrzelak) [inline Module.yaml]: please start the words with capital letters as it looks better in UI

File: README.md [hunk 6]
```diff
@@ -1,2 +1,73 @@
...
+### Set by the weeve Agent on the edge-node
+
+| Environment Variables | type   | Description               |
+| --------------------- | ------ | ------------------------- |
+| MODULE_NAME           | string | Name of the module        |
+| INGRESS_HOST          | string | Host where app is running |
+| INGRESS_PORT          | string | Port where app is running |
+
+
+
```
Comment (kubagrzelak) [inline README.md]: Is there any Input example ?

File: deploy.env [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# ENV VARIABLES FOR DEPLOYMENT
+# You have to define the values in {}
+APP_NAME=opcua
```
Comment (kubagrzelak) [inline deploy.env]: This should be the container name so: `opcua-subscribe`

File: image/entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+# More safety, by turning some bugs into errors.
+# Without `errexit` you don’t need ! and can replace
+# PIPESTATUS with a simple $?, but I don’t do that.
+set -o errexit -o pipefail -o noclobber -o nounset
+
+echo "Entrypoint script for opcua client"
```
Comment (kubagrzelak) [inline image/entrypoint.sh]: I think instead of `opcua client` it should be `opcua-subscribe`

**Review by kubagrzelak — CHANGES_REQUESTED** [review]

File: Module.yaml (hunk 4 above)
Comment (mesudpasic) [reply Module.yaml]: module acts as standalone, listens for events and forms a JSON and sends it over to next module, something similar like scheduler.

File: Module.yaml (hunk 3 above)
Comment (mesudpasic) [reply Module.yaml]: ok I can update it, but I believe we used process in the past with other modules.

File: Module.yaml (hunk 1 above)
Comment (mesudpasic) [reply Module.yaml]: updated.

File: Module.yaml (hunk 5 above)
Comment (mesudpasic) [reply Module.yaml]: updated, my bad :)

File: README.md (hunk 6 above)
Comment (mesudpasic) [reply README.md]: no, there's no input for this module.

File: deploy.env (hunk 7 above)
Comment (mesudpasic) [reply deploy.env]: updated

File: image/entrypoint.sh (hunk 8 above)
Comment (mesudpasic) [reply image/entrypoint.sh]: updated

File: Module.yaml (hunk 2 above)
Comment (mesudpasic) [reply Module.yaml]: requested

File: Module.yaml [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+      - no
+  - key:  OPC_UA_USERNAME
+    description: Username for authentication
+    value: ''
+    default: ''
+    type: string
+    name: OPC UA Username
+    dependencies:
+      - key: REQUIRES_AUTHENTICATION
+        value: yes
```
Comment (kubagrzelak) [inline Module.yaml]: `dependencies` field is not supported in V0 (we support in V1). This module is currently in V0 so please remove it.

File: Module.yaml [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+  - key:  OPC_UA_PASSWORD
+    description: Password for authentication
+    value: ''
+    default: ''
+    type: string
+    name: OPC UA Password
+    dependencies:
+      - key: REQUIRES_AUTHENTICATION
+        value: yes
```
Comment (kubagrzelak) [inline Module.yaml]: Same

File: Module.yaml (hunk 9 above)
Comment (mesudpasic) [reply Module.yaml]: removed.

File: Module.yaml (hunk 10 above)
Comment (mesudpasic) [reply Module.yaml]: removed.

File: Module.yaml (hunk 2 above)
Comment (mesudpasic) [reply Module.yaml]: module is changed to input based on our last call.

**Review by nithinsaii — APPROVED** [review]
Module.yaml looks good and container runs fine locally.

**Pull request merged**